src/prognostic_equations/limited_tendencies.jl

```diff
@@ -1,32 +1,23 @@
-using LinearAlgebra: ×, norm, dot
-
-import .Parameters as CAP
-using ClimaCore: Operators, Fields, Limiters, Geometry, Spaces
-
-import ClimaComms
-using ClimaCore.Geometry: ⊗
-
-import Thermodynamics as TD
-
-using ClimaCore.Utilities: half
-
-import ClimaCore.Fields: ColumnField
-
 NVTX.@annotate function limited_tendency!(Yₜ, Y, p, t)
     Yₜ .= zero(eltype(Yₜ))
     horizontal_tracer_advection_tendency!(Yₜ, Y, p, t)
     tracer_hyperdiffusion_tendency!(Yₜ, Y, p, t)
     return nothing
 end
 
-NVTX.@annotate function limiters_func!(Y, p, t, ref_Y)
-    (; limiter) = p
-    n = n_mass_flux_subdomains(p.atmos.turbconv_model)
-    if !isnothing(limiter)
+function generate_limiters_func!(limiter::Nothing)
+    limiters_func! = (Y, p, t, ref_Y) -> nothing
+    return limiters_func!
+end
+
+function generate_limiters_func!(limiter)
```

Comment:
I prefer we avoid this closure. What is this needed for?

Reply:
`limiters_func!` has a specific signature. If I want to dispatch over types of limiters (instead of checking the boolean `apply_limiter`, I need to return it as a function. This allows me to remove `limiter` from the cache, as `limiter` is used exclusively here.

Reply:
Couldn't we instead change `limiters_func!(Y, p, t, ref_Y)` to `limiters_func!(::QuasiMonotoneLimiter, Y, p, t, ref_Y)` and define
```julia
limiters_func!(Y, p, t, ref_Y) = limiters_func!(p.limiter, Y, p, t, ref_Y)
limiters_func!(::Nothing, args...) = nothing
```
?
This would require fewer changes across the codebase, too.

Reply:
And why do we need to remove limiter from the cache?

Reply:
> And why do we need to remove limiter from the cache?

Because the limiter is used exclusively in `limiters_func!`, so if we want to reduce the variables we carry around in the cache, this is a good place to do that. But we need to capture `limiter` to do that.

Reply:
We can still reduce variables in the cache without removing this one. I'd prefer we leave this in the cache-- one disadvantage of doing this is that it becomes difficult to interactively inspect the `QuasiMonotoneLimiter` object (which can be helpful for debugging), since it only exists in the anonymous struct generated by the closure.

Reply:
@kmdeck, is this something that you've found in ClimaLSM? (I think this pattern may be pretty heavily used there)

Reply:
Okay, I don't have strong feelings about this.

Reply:
Rebased and fixed this. I put the limiter inside `p.numerics`.

Reply:
@charleskawczynski let me know if I am not answering your question...

we use closures like this a lot:
```
function make_update_aux(model::MyModelType)
    function update_aux!(p,Y,t)
        #update elements in p in place based on type of `model` (simple dispatch) but also
        # using parameters, etc that are stored in `model` (closure), Y, p
    end
    return update_aux!
end
```

I agree that they are kind of annoying to write tests for because you need to make the `model` in order to test it. but we also need to contruct Y and p with the necessary structure to debug/write tests for, so the closure isnt the only annoying part.

Reply:
and, I guess the alternative would be to pass `model` as an argument, which also doesnt get around having to make a `MyModelType` instance